module-2/lab-probability-distribution/your_code/main.ipynb

```diff
@@ -366,7 +366,7 @@
   },
```

Comment:
Sometimes it's useful to seed the random number generator. Seeding a pseudo-random number generator gives it its first "previous" value. Each seed value will correspond to a sequence of generated values for a given random number generator. That is, if you provide the same seed twice, you get the same sequence of numbers twice. https://machinelearningmastery.com/how-to-generate-random-numbers-in-python/
-Challenge 1 - Generate and Plot Normal Distributions: Try to do this challenge in a single line for all sample sizes. Figure it out how to transform the procedural you have done into a more structured chunk of code.
- Normality tests are quite sensitive to sample size thus for a sample size below a couple of thousands it should be a better idea to evaluate normality through a plot, in particular a quantile-quantile plot (QQ plot). A Q-Q plot is a scatterplot created by plotting two sets of quantiles against one another. It plots the quantiles of our sample vs the theoretical quantiles of the distribution type we want to test (this case a normal distribution) How to do this in Python? https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.probplot.html.
- There are 2 other normality tests (additional to D'Agostino k^2 test, normaltest in Python) that are very used in to test normality of our data. These are Kolmogorov-Smirnof and Shapiro-Wilk tests. I encourage you to deepen on these type of tests. In case you are interested, I can provide you some useful resources.
- Interpretation of the CLT: take a look on these 2 articles (the most clear explanations I've found). You can replicate in Python some of the examples they include.
https://towardsdatascience.com/understanding-the-central-limit-theorem-642473c63ad8.
- Try to do the following chunk of code  in a couple of lines. 
```Python
p = 0.15
n_50 = 50
k_50 = range(0,n_50+1)
n_500 = 500
k_500 = range(0,n_500+1)
n_5000 = 5000
k_5000 = range(0,n_5000+1)
binomial_dist_50 = sci.binom.pmf(k_50, n_50, p)
binomial_dist_500 = sci.binom.pmf(k_500, n_500, p)
binomial_dist_5000 = sci.binom.pmf(k_5000, n_5000, p)
fig, ax = plt.subplots(3,1, figsize=(12,6))
ax[0].plot(k_50, binomial_dist_50)
ax[0].set_xlabel('number of click')
ax[0].set_ylabel('probability of click')
ax[0].set_title('n = 50')
ax[1].plot(k_500, binomial_dist_500)
ax[1].set_xlabel('number of clicks')
ax[1].set_ylabel('probability of clicks')
ax[1].set_title('n = 500')
ax[2].plot(k_5000, binomial_dist_5000)
ax[2].set_xlabel('number of clicks')
ax[2].set_ylabel('probability of clicks')
ax[2].set_title('n = 5000')
```
- This is not quite correct on Binomial challenge : "What did you notice from the distribution plots? Comment your findings.
As the number of visitors increases, the probability of them clicking becomes close to zero". 
Normal distribution can be used as an approximation of Binomial and Poisson distributions under certain circumstances. Generally speaking:
Binomial: X ~ B(n, p) and if n is large and/or p is close to ½, then X is approximately N(np, npq)
Poisson: X ~ Po(l) then for large values of l, X ~ N(l, l) approximately.
However, we must be careful because binomial and poisson distributions are discrete while Normal distribution is continuous. We must use a continuity correction. 
- One way for challenge 3:
```Python

# your code here
fig, ax = plt.subplots(1, 1)
mu=300
x = np.arange(0,1000)
ax.plot(x, sci.poisson.pmf(x, mu), 'bo', ms=8, label='poisson pmf')
```
- Try to set x and y labels as well as the title when plotting. It's a very good practice.